### ISSUE TITLE
Change default scaleFactor to 10

### ISSUE THREAD

**Issue by FlorianWetschoreck** [vega/vega-embed#482]
Hi,

when currently exporting Vega Charts to PNG using vega embed, the result is usually pixelated because the default of the scaleFactor is 1. I achieved satisfactory results out of the box when using a scaleFactor of 10.

Thus, I propose to change the default scaleFactor to 10 which can be still overriden by the user, of course.

This would also result in favorable downstream effects: e.g. PNG chart exports using Altair in Jupyter could be done via the inline rendered vega embed instead of always having to go through a complicated mix of steps - either exporting to SVG and then trying to convert this to PNG or setting up the other dependencies of altair_saver etc

- Are there any side effects that might prevent this change?

I am also open to provide a PR in case that the proposal is accepted

**Comment by domoritz** [comment]
The issue is that this would increase image sizes. I would say we can increase the default to 2, which would match 2x screen resolutions.

Note that you can override the embed options for altair as well by setting the `usermeta` property.

**Comment by FlorianWetschoreck** [comment]
Thank you for your comment - and the hint with the usermeta property was very helpful to me. Both in a pure Vega context as well as for Altair.

Here is my personal user story:
- I intend to use the PNG export, when I want to embed the graph into a PowerPoint presentation or Latex document OR send it to someone else who is uncomfortable opening/receiving svg or html files. In this scenario, I want the PNG to be really crisp out of the box. I don't care so much about the exact dimensions of the graph because I can usually scale the image eg in PowerPoint or Latex. Also, the other person does not care so much because the actual size depends on the software that he uses on his computer to open the graph. The alternative to just hitting "Download as PNG" is to go through the other steps of downloading the SVG and then using another program to convert this to PNG which is more effort. So, for me this is mostly about saving time of getting a crisp PNG.
- When I am able to quickly get a crisp PNG out of the box, I am excited. Receiving the pixelated PNG I was disappointed that I have to do a cumbersome dance of exporting the SVG and then converting etc. However, I appreciate the option of accessing the SVG which gives me at least flexibility to achieve what I want to achieve given enough technical effort.
- Now that I know about the usermeta property, I would default to always adding the usermeta for a scaleFactor of 10x if the default is 1x. I would prefer to not having to think about this and would be excited if this comes out of the box. Having to add usermeta I would feel indifferent as long as I can always quickly find the fix and add it quickly although this might become a little bit bothersome when having to export many graphs over and over.

About the alternatives:
- I experimented with a graph that had original dimensions of 200x200 as a bottom line (as this is similar to many cases in Jupyter and also to many of the Vega and Altair examples) - and the minimum satisfactory result for me was a scaleFactor of 5. However, I prefer the additional crispness of a scaleFactor of 10. In total, I tried, 1, 2, 3, 5 and 10.
- I am not so sure if by image size you referred to the width and height or the file size in KB. When I tried to adjust my graph to a 10x size in order to "hack" the scaleFactor (in case I am a user who does not know about scaleFactor and usermeta), I was disappointed because this totally changes the aesthetics of the graph with regard to text size etc. For me all the file sizes were fine as this is not usually a concern.
- In the past, I sometimes zoomed into graphs and made "high resolution" screenshots but this approach is not so great because it is not repeatable.
- If I were a user who cares about setting the actual size via width and height, I guess it would be less confusing when the sizes are 10x instead of 2x, 3x, or 5x.

Summing up, changing the scaleFactor to 2x would make no change to me. I would feel happy with a scaleFactor of 5x and excited with a scaleFactor of 10x.
In the end it is a tradeoff between "being correct" and "being exciting in the above scenario". When I have to weigh this, I would strongly prefer excitement. Also, I think that the "correct scenario" is way less common and for the people who want to be correct, they can still do this via SVG and the usual workaround.

I am looking forward to your final decision and how you weigh the tradeoffs

**Comment by domoritz** [comment]
Thank you for writing up the user story and explaining why 2x isn't enough.

For latex and presentations, I recommend that you use PDF instead as they are often smaller and scale arbitrarily. Here is a guide for generating PDFs for papers: https://vega.github.io/vega-lite/tutorials/figures.html. The editor also has PDF download build in (we may add this to embed at some point).

<img width="646" alt="Screen Shot 2020-06-21 at 10 13 03" src="https://user-images.githubusercontent.com/589034/85230876-d99bd400-b3a7-11ea-9695-efd70ab65480.png">

For all other cases where you need high-resolution PNGs, you can set the usermeta property. 

I am not going to change the default right now since images would get larger (in terms of kb) and I haven't heard other people ask for a different default yet. 

As with any design decision, thank you for the discussion and we may revisit this decision in the future.
